src/beneuro_data/conversion/multiprobe_kilosort_interface.py

```diff
@@ -1,13 +1,125 @@
 from pathlib import Path
 from typing import Literal, Optional
 
 import numpy as np
+import pandas as pd
 from neuroconv.datainterfaces import KiloSortSortingInterface
 from neuroconv.tools.spikeinterface import add_sorting_to_nwbfile
 from neuroconv.utils import DeepDict
 from pydantic import DirectoryPath
 from pynwb import NWBFile
+import probeinterface as pi
+
+
+def _try_loading_trajectory_file(raw_recording_path):
+    """
+    Load trajectory of probes from pinpoint output files
+
+    Parameters
+    ----------
+    raw_recording_path :
+        path to raw session
+
+    Returns
+    -------
+    Dictionary with trajectory information per probe
+
+    """
+
+    pinpoint_trajectory_file = list(raw_recording_path.glob("*trajectory.txt"))
+    if not pinpoint_trajectory_file:
+        warnings.warn(
+            f"No trajectory file from pinpoint found. If no trajectory file is present"
+            f" channel map information cannot be loaded"
+        )
+        return
+
+    elif len(pinpoint_trajectory_file) > 1:
+        raise FileExistsError("Too many files found; expected only one.")
+
+    with open(pinpoint_trajectory_file[0], "r") as f:
+        trajectory_str = [l.strip() for l in f.readlines() if l.strip() != ""]
+        probe_trajectory_pairs = list(zip(trajectory_str[::2], trajectory_str[1::2]))
+        trajectory_dict = {
+            probe: trajectory for probe, trajectory in probe_trajectory_pairs
+        }
+
+    return trajectory_dict
+
+
+def _load_channel_map_information_from_pinpoint_probe(filename, pinpoint_probe_name):
+    """
+    Extract the brain area of each electrode per probe
+
+    Parameters
+    ----------
+    filename :
+        Pinpoint channel_map file to open
+    pinpoint_probe_name :
+        Key generated in pinpoint to identify the probe
+
+    Returns
+    -------
+    Dataframe containing bran area of each electrode
+
+    """
+    with open(filename, "r") as file:
+        channel_map_str = file.read().strip()
+
+    channel_map_str = channel_map_str.strip("[]").split('","')
+    data_string = ""
+    for item in channel_map_str:
+        if pinpoint_probe_name in item:
+            data_string = item.split(":", 1)[1]
+            break
+
+    # Split the string by ";" to separate each entry
+    entries = data_string.split(";")
+    data = [
+        dict(zip(["id", "area_number", "area_name", "area_color"], entry.split(",")))
+        for entry in entries
+    ]
+
+    df = pd.DataFrame(data)
+    return df
+
+
+def _create_channel_map(location_information, raw_recording_path):
+    """
+    Function to create the channel map of each probe
+
+    Parameters
+    ----------
+    location_information :
+        Pinpoint generate trajectory file (*trajectory.txt)
+    raw_recording_path :
+        Path to raw session
+
+    Returns
+    -------
+    Dictionary of channel map for each probe
+
+    """
+
+    pinpoint_trajectory_file = list(raw_recording_path.glob("*channel_map.txt"))
```

Comment:
This should be called something like `pinpoint_channel_map_file`, right? I was very confused when reading it.

Reply:
This was a poor attempt at not mixing channel_map variables, because kilosort also has a channel map. I changed it to your suggestion